src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -137,87 +140,90 @@ class DuplexConsensusCaller(override val readNamePrefix: String,
     val (pairs, frags) = recs.partition(_.paired)
     rejectRecords(frags, FilterFragments)
 
-    // Group the reads by /A vs. /B and ensure that /A is the first group and /B the second
-    val groups = pairs.groupBy(r => r[String](ConsensusTags.MolecularId)).toSeq.sortBy { case (mi, _) => mi }.map(_._2)
-
-    groups match {
-      case Seq() =>
-        Nil
-      case Seq(ab) =>
-        rejectRecords(ab, FilterAorBOnly)
-        Nil
-      case Seq(ab, ba) =>
-        // Fragments have no place in duplex land (and are filtered out previously anyway)!
-        val (_, abR1s, abR2s) = subGroupRecords(ab)
-        val (_, baR1s, baR2s) = subGroupRecords(ba)
-
-        // Get all the alignments to one end of the source molecule
-        val singleStrand1 = abR1s ++ baR2s
-        val singleStrand2 = abR2s ++ baR1s
-
-        // The orientation of AB and BA reads should be:
-        // AB R1: +  AB R2: -
-        // BA R1: -  BA R2: +
-        // or vice versa (AB-R1:-, AB-R2:+, AB-R1:-, AB-R2: +
-        // Therefore, AB-R1s and BA-R2s should be on the same strand, and the same for AB-R2s and BA-R1s
-        // Check for this explicitly here.
-        if (singleStrand1.nonEmpty) {
-          val ss1Flag = singleStrand1.head.negativeStrand
-          if (singleStrand1.exists(_.negativeStrand != ss1Flag)) {
-            val ss1MI   = singleStrand1.head.apply[String](ConsensusTags.MolecularId)
-            rejectRecords(ab ++ ba, FilterCollision)
-            logger.debug(s"Not all AB-R1s and BA-R2s were on the same strand for molecule with id: $ss1MI")
-            return Nil
-          }
-        }
-        if (singleStrand2.nonEmpty) {
-          val ss2Flag = singleStrand2.head.negativeStrand
-          if (singleStrand2.exists(_.negativeStrand != ss2Flag)) {
-            val ss2MI   = singleStrand2.head.apply[String](ConsensusTags.MolecularId)
-            rejectRecords(ab ++ ba, FilterCollision)
-            logger.debug(s"Not all AB-R2s and BA-R1s were on the same strand for molecule with id: $ss2MI")
-            return Nil
-          }
-        }
+    // Partition reads into those that end with /A and those that end with /B (verify the latter
+    val (ab, ba) = pairs.partition(r => r[String](ConsensusTags.MolecularId).endsWith("/A"))
```

Comment:
I don't like how this and the subsequent lines encode direct knowledge of the expected MI structure.  The only requirement elsewhere is that the MI be of the form "common-prefix/one-of-two-suffixes".  There's even a method in this class `def sourceMoleculeId(rec: SamRecord): String` to get the molecule ID that is used elsewhere.

Reply:
If you really need to know AB vs. BA (not just one vs. two sets) then I think we need `GroupReadsByUmi` to expose the two suffixes as public vals and re-use them here.